data-connection/src/main/java/io/micronaut/data/connection/interceptor/ConnectableInterceptor.java

```diff
@@ -150,20 +161,47 @@ public Object intercept(MethodInvocationContext<Object, Object> context) {
     }
 
     @NonNull
-    public static ConnectionDefinition getConnectionDefinition(ExecutableMethod<Object, Object> executableMethod) {
+    public static ConnectionDefinition getConnectionDefinition(ExecutableMethod<Object, Object> executableMethod, String appName) {
         AnnotationValue<Connectable> annotation = executableMethod.getAnnotation(Connectable.class);
         if (annotation == null) {
             throw new IllegalStateException("No declared @Connectable annotation present");
         }
 
+        ConnectionClientTracingInfo connectionClientInformation = getConnectionClientTracingInfo(annotation, executableMethod, appName);
         return new DefaultConnectionDefinition(
             executableMethod.getDeclaringType().getSimpleName() + "." + executableMethod.getMethodName(),
             annotation.enumValue("propagation", ConnectionDefinition.Propagation.class).orElse(ConnectionDefinition.PROPAGATION_DEFAULT),
             annotation.longValue("timeout").stream().mapToObj(Duration::ofSeconds).findFirst().orElse(null),
-            annotation.booleanValue("readOnly").orElse(null)
+            annotation.booleanValue("readOnly").orElse(null),
+            connectionClientInformation
         );
     }
 
+    /**
+     *
+     * @param annotation
+     * @param executableMethod
+     * @param appName
+     * @return
+     */
+    private static ConnectionClientTracingInfo getConnectionClientTracingInfo(AnnotationValue<Connectable> annotation,
+                                                                              ExecutableMethod<Object, Object> executableMethod,
+                                                                              String appName) {
+        boolean traceClientInfo = annotation.booleanValue(TRACE_CLIENT_INFO_MEMBER).orElse(false);
+        if (!traceClientInfo) {
+            return null;
+        }
+        String module = annotation.stringValue(TRACING_MODULE_MEMBER).orElse(null);
+        String action = annotation.stringValue(TRACING_ACTION_MEMBER).orElse(null);
+        if (module == null) {
+            module = executableMethod.getDeclaringType().getName();
```

Comment:
This is picking up parent class/interface when used on Micronaut Data repository class (interface) (so in one example it returns `io.micronaut.data.repository.CrudRepository` instead of expected `com.example.BookRepository`
When `@Connectable` is added to the method then it finds correct class/interface name.

Reply:
When annotation `Connectable` (or if we add some other) is not added on the method then method `findById` being executed is from the `CrudRepository` interface and we can't get actual interface `BookRepository`. That means we might have to provide module name (required) to the repository class via annotation or else we can't have actual repo class.

Reply:
Here you have to resolve the type from `context.getTarget().getClass()` and likely get the super interface/class of that type.